### PR TITLE
[ICP-13163] Add alarm methods to child, expose chime sounds

### PR DESCRIPTION
**Pull request by MGoralczykS**
@greens @dkirker 
Could you check it? 
Sounds from 3 to 5 are chimes. 

cc: @KKlimczukS @PKacprowiczS @ZWozniakS @BJanuszS @MWierzbinskaS

### CODE REVIEW

File: devicetypes/smartthings/aeotec-doorbell-siren-6.src/aeotec-doorbell-siren-6.groovy [hunk 1]
```diff
@@ -244,13 +244,16 @@ def resetActiveSound() {
...
 }
 
 def setActiveSound(soundId) {
 	String childDni = "${device.deviceNetworkId}:${soundId}"
 	def child = childDevices.find { it.deviceNetworkId == childDni }
-	child?.sendEvent([name: "chime", value: "chime"])
+	if(soundId >= 3 || soundId <= 5) {
+		child?.sendEvent(name: "chime", value: "on", displayed: false)
+	} else {
+		child?.sendEvent(name: "alarm", value: "both", displayed: false)
```
Comment (greens) [inline devicetypes/smartthings/aeotec-doorbell-siren-6.src/aeotec-doorbell-siren-6.groovy]: unless you're sending both a "charm" and an "alarm" event for every device I think you'll still see the network error.

File: devicetypes/smartthings/aeotec-doorbell-siren-6.src/aeotec-doorbell-siren-6.groovy [hunk 2]
```diff
@@ -225,8 +225,8 @@ def setOnChild(deviceDni) {
 	resetActiveSound()
 	sendHubCommand encap(zwave.basicV1.basicSet(value: 0xFF), channelNumber(deviceDni))
 	state.lastTriggeredSound = channelNumber(deviceDni)
-	setActiveSound(state.lastTriggeredSound)
-}
+	setActiveSound(state.lastTriggeredSound) 
+	}
```
Comment (dkirker) [inline devicetypes/smartthings/aeotec-doorbell-siren-6.src/aeotec-doorbell-siren-6.groovy]: Spacing

File: devicetypes/smartthings/aeotec-doorbell-siren-6.src/aeotec-doorbell-siren-6.groovy (hunk 1 above)
Comment (dkirker) [reply devicetypes/smartthings/aeotec-doorbell-siren-6.src/aeotec-doorbell-siren-6.groovy]: Yeah, you're going to need to send both events, or do one of my suggestions in the original creation and have two different child device types; one for doorbell and one for siren. That might be hard to do now that the device has already been created. Though you *might* be able to re-type child devices (keep in mind if anyone has managed to create rules based on those, they would likely break).

File: devicetypes/smartthings/aeotec-doorbell-siren-6.src/aeotec-doorbell-siren-6.groovy [hunk 3]
```diff
@@ -225,8 +225,8 @@ def setOnChild(deviceDni) {
 	resetActiveSound()
 	sendHubCommand encap(zwave.basicV1.basicSet(value: 0xFF), channelNumber(deviceDni))
 	state.lastTriggeredSound = channelNumber(deviceDni)
-	setActiveSound(state.lastTriggeredSound)
-}
+	setActiveSound(state.lastTriggeredSound)
```
Comment (dkirker) [inline devicetypes/smartthings/aeotec-doorbell-siren-6.src/aeotec-doorbell-siren-6.groovy]: Extra space at end of line

**Comment by MGoralczykS** [comment]
@greens @dkirker Could you check it?

File: devicetypes/smartthings/aeotec-doorbell-siren-child.src/aeotec-doorbell-siren-child.groovy [hunk 4]
```diff
@@ -26,12 +26,15 @@ metadata {
 				attributeState "chime", label: 'off', action: 'chime.off', icon: "st.alarm.alarm.alarm", backgroundColor: "#ff0000"
 			}
 		}
+		standardTile("off", "device.alarm", inactiveLabel: false, decoration: "flat") {
```
Comment (dkirker) [inline devicetypes/smartthings/aeotec-doorbell-siren-child.src/aeotec-doorbell-siren-child.groovy]: Is the purpose of this to provide an "off" command? If so, you already have the one for chime off, which should operate the same command. You might just want to change this to a value tile (and also rename it to "alarm" as you currently have two "off" tiles), or remove it altogether, as both chime and alarm events are sent to the child device.

File: devicetypes/smartthings/aeotec-doorbell-siren-6.src/aeotec-doorbell-siren-6.groovy [hunk 5]
```diff
@@ -244,13 +244,13 @@ def resetActiveSound() {
...
 	sendEvent([name: "alarm", value: "off"])
-	sendEvent([name: "chime", value: "off"])
 }
 
 def setActiveSound(soundId) {
 	String childDni = "${device.deviceNetworkId}:${soundId}"
 	def child = childDevices.find { it.deviceNetworkId == childDni }
-	child?.sendEvent([name: "chime", value: "chime"])
+	child?.sendEvent(name: "chime", value: "on", displayed: false)
+	child?.sendEvent(name: "alarm", value: "both", displayed: false)
```
Comment (dkirker) [inline devicetypes/smartthings/aeotec-doorbell-siren-6.src/aeotec-doorbell-siren-6.groovy]: Was the `displayed: false` intentional?  If you do this, only the "off" events will ever show up.

File: devicetypes/smartthings/aeotec-doorbell-siren-6.src/aeotec-doorbell-siren-6.groovy [hunk 6]
```diff
@@ -244,13 +244,13 @@ def resetActiveSound() {
 		sendHubCommand(onOffCmd(0x00))
 	}
 	sendEvent([name: "alarm", value: "off"])
-	sendEvent([name: "chime", value: "off"])
```
Comment (dkirker) [inline devicetypes/smartthings/aeotec-doorbell-siren-6.src/aeotec-doorbell-siren-6.groovy]: Why did you remove this? The parent DTH also has the Chime capability.

You are also missing events for chime on lines 155 and 207.

File: devicetypes/smartthings/aeotec-doorbell-siren-6.src/aeotec-doorbell-siren-6.groovy (hunk 5 above)
Comment (dkirker) [reply devicetypes/smartthings/aeotec-doorbell-siren-6.src/aeotec-doorbell-siren-6.groovy]: The `displayed: false` only makes sense on initialization, in `keepChildrenOnline()`, and probably for the parent events.

**Review by dkirker — CHANGES_REQUESTED** [review]

**Comment by dkirker** [comment]
I guess I had missed a couple of things the first time around. There are a few things that I have noted, including a couple of lines from the original DTH that might cause a problem.

**Comment by MGoralczykS** [comment]
@dkirker Could you check it?

**Comment by dkirker** [comment]
@MGoralczykS I believe you still need equivalent chime events for the alarm events on lines 155 and 209. I also believe that you probably want the even on line 155 to not have `displayed: false`?

Also, please see my comment in aeotec-doorbell-siren-child.groovy.

**Comment by MGoralczykS** [comment]
@dkirker Could you check it? I've made changes in aeotec-doorbell-siren-child is this as you wanted? Or I misunderstood you?

File: devicetypes/smartthings/aeotec-doorbell-siren-child.src/aeotec-doorbell-siren-child.groovy [hunk 7]
```diff
@@ -20,15 +20,18 @@ metadata {
...
+		multiAttributeTile(name: "chime", type: "generic", width: 6, height: 2) {
 			tileAttribute("device.chime", key: "PRIMARY_CONTROL") {
 				attributeState "off", label: 'chime', action: 'chime.chime', icon: "st.alarm.alarm.alarm", backgroundColor: "#ffffff"
 				attributeState "chime", label: 'off', action: 'chime.off', icon: "st.alarm.alarm.alarm", backgroundColor: "#ff0000"
 			}
 		}
-		standardTile("off", "device.alarm", inactiveLabel: false, decoration: "flat") {
-			state "default", label: '', action: "alarm.off", icon: "st.secondary.off"
-		}
+		multiAttributeTile(name: "alarm", type: "generic", width: 6, height: 2) {
```
Comment (dkirker) [inline devicetypes/smartthings/aeotec-doorbell-siren-child.src/aeotec-doorbell-siren-child.groovy:29]: Do two multi-attr tiles actually work??

Comment (MGoralczykS) [reply devicetypes/smartthings/aeotec-doorbell-siren-child.src/aeotec-doorbell-siren-child.groovy:29]: Yes, it does but there are two large buttons.

**Comment by greens** [comment]
going to wait on @dkirker for this one

File: devicetypes/smartthings/aeotec-doorbell-siren-child.src/aeotec-doorbell-siren-child.groovy (hunk 7 above)
Comment (dkirker) [reply devicetypes/smartthings/aeotec-doorbell-siren-child.src/aeotec-doorbell-siren-child.groovy:29]: Ah yes! So it does! I actually had never thought to try it. (Though I do see it in the docs, too.)

**Review by dkirker — APPROVED** [review]
Awesome! Thanks @MGoralczykS !
